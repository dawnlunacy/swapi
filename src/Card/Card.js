
import React from 'react';
import { Link } from 'react-router-dom';
import './Card.scss'

const Card = ({ data, findCharacters}) => {
    console.log("CardProp",data)
    // console.log("findCharacters", findCharacters)
    // const { date } = new Date(data.release_date + 'T00:00').toString().split(' ').slice(1,4 ).join(' ');
    const { title, episode_id, release_date, name, homeWorld, homeWorldPopulation, species, films } = data;
    // const film = films.map(film => <li> {film }</li>)

    return(
        <div className="card">
            <header className="card-header">
            {/* <h2> {data.title} </h2>
            <h2> {data.name} </h2> */}
            
            { title && <h2> { title } </h2>}
            { name && <h2> { name }</h2> }

            </header>
            <main className="card-main">
            {/* <h3>  {data.species}</h3>
            <h3>  {data.homeWorld}</h3> */}
             {/* <h3>  { species }</h3>
            <h3>  { homeWorld }</h3>
            <h3> { homeWorldPopulation }</h3>
            <h3>  {episode_id}</h3> */}
        {/* <ul> { films.map(film => <li> { film } </li>) } </ul> */}
            <ul> 
            { episode_id && <li> <h3> Episode { episode_id }</h3></li>}
            { release_date && <li><h3> { new Date(data.release_date + 'T00:00').toString().split(' ').slice(1,4 ).join(' ') } </h3></li>}

            { species && <li> Species: {species} </li>}
            { homeWorld && <li> Home World: {homeWorld} </li>}
            { homeWorldPopulation && <li> Population of Home World: {homeWorldPopulation} </li>}
            { species && <li> Species: {species} </li>}
            { films && films.map(film => <li> { film.title } </li>)}
            </ul>

            {/* <h3>{new Date(release_date + 'T00:00').toString().split(' ').slice(1,4 ).join(' ')}</h3> */}
            </main>
            <footer>
            {/* <button onClick={event => findCharacters(data.characters)}> View Characters </button> */}
            <Link to='/movies/moose' className='btn'>  
            { title && <button onClick={event => findCharacters(data.characters)}> View Characters </button> }            
             </Link>
            </footer>
        </div>
        
    )
}

export default Card;



