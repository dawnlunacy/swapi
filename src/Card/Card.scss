.card {
    border: solid 2px black;
    display: block;
    background-color: teal;
}