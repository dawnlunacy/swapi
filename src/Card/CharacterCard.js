
import React from 'react';
import { Link } from 'react-router-dom';
import './Card.scss'

const Card = ({ data, findCharacters}) => {
    // console.log("CardProp",data)
    // console.log("findCharacters", findCharacters)

    return(
        <div className="card">
        <header className="card-header">
        <h2> {data.name} </h2>

        </header>
        <main className="card-main">
        <h3>  {data.species}</h3>

        </main>
        <footer>
        {/* <button onClick={event => findCharacters(data.characters)}> View Characters </button> */}
        <Link to='/movies/moose' className='btn'>  
        <button onClick={event => findCharacters(data.characters)}> View Characters </button>            
         </Link>

        </footer>
    </div>

    )
}

export default Card;