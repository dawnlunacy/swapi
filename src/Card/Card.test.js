import React from 'react';
import { shallow } from 'enzyme';
import Card from '../Card/Card';

describe('Card', () => {
    let wrapper
    const mockFindCharacters = jest.fn()
    beforeEach(() => {
        wrapper = shallow(<Card
            className="movie-card"
            key={2}
            data={[{title: 'Attack of the Clones', episode_id: 2, opening_crawl: 'There is unrest in the Galactic...', release_date: '2002-05-16', characters: Array(2)}]}
            findCharacters={jest.fn()}
            />)
    });
    it('should match snapshot with all the correct data passing through', () => {
        expect(wrapper).toMatchSnapshot()
    });
    it.skip('should run findCharacters when View Characters is clicked', () => {
        wrapper.find('button').simulate('click')

        expect(mockFindCharacters).toHaveBeenCalledWith(2);
    }) // this test will complete the testing for card coverage
})