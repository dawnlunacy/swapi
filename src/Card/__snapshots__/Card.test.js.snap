// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot with all the correct data passing through 1`] = `
<div
  className="card"
>
  <header
    className="card-header"
  >
    <h2>
       
      Attack of the Clones
       
    </h2>
  </header>
  <main
    className="card-main"
  >
    <ul>
      <li>
         
        <h3>
           Episode 
          2
        </h3>
      </li>
      <li>
        <h3>
           
          May 16 2002
           
        </h3>
      </li>
    </ul>
  </main>
  <footer>
    <Link
      className="btn"
      to="/movies/2"
    >
      <button
        onClick={[Function]}
      >
         View Characters 
      </button>
    </Link>
  </footer>
</div>
`;
